internal/operator-controller/authorization/rbac.go

```diff
@@ -112,7 +107,20 @@ func (a *rbacPreAuthorizer) PreAuthorize(ctx context.Context, manifestManager us
 		if compactMissingRules, err := validation.CompactRules(nsMissingRules); err == nil {
 			missingRules[ns] = compactMissingRules
 		}
-		sortableRules := rbacv1helpers.SortableRuleSlice(missingRules[ns])
+
+		missingRulesWithDeduplicatedVerbs := []rbacv1.PolicyRule{}
+		for _, rule := range missingRules[ns] {
+			verbSet := sets.New[string](rule.Verbs...)
+			if verbSet.Has("*") {
+				rule.Verbs = []string{"*"}
+			} else {
+				rule.Verbs = sets.List(verbSet)
+			}
+			missingRulesWithDeduplicatedVerbs = append(missingRulesWithDeduplicatedVerbs, rule)
+		}
+
+		sortableRules := rbacv1helpers.SortableRuleSlice(missingRulesWithDeduplicatedVerbs)
```

Comment:
Should result in deduped verbs in the rules

Reply:
are we using rbacv1helpers.SortableRuleSlice for parity with kubernetes code or for convenience?

if the latter, I'd suggest:

```
slices.SortFunc(missingRulesWithDeduplicatedVerbs, func(a, b rbacv1.PolicyRule) int {
        return cmp.Compare(a.String(), b.String())
})
```

because it sames me from having to look into rbacv1helpers.SortableRuleSlice to understand the sort order XD

Reply:
Parity and to make sure we don't do it incorrectly.

Reply:
@perdasilva so if you're okay with this answer then I'll resolve this thread?